File: reverse_integer/reverse_int.py
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            x = -int(str(-x)[::-1]) # use s[::-1] to reverse string s
        else:
            x = int(str(x)[::-1])
            
        if(x < -2**31 or x > 2**31):
            return 0
            
        return x

File: reverse_integer/reverse_int.cpp
class Solution {
public:
    int reverse(int x) {
        if (x < INT_MIN || x > INT_MAX) return 0;
        string s = to_string(x); // convert int to string
        std::reverse(s.begin(), s.end()); // reverse the string s
        long c = stol(s); // convert string to long type
        if (c > INT_MAX) return 0;
        return x < 0 ? (0-c) : c;
    }
};
